### PR TITLE
user presence event

### PR DESCRIPTION
**Pull request by Connoropolous**

### CODE REVIEW

**Comment by devvmh** [comment]
is this in progress for reals?

File: app/assets/javascripts/src/Metamaps.Visualize.js [hunk 1]
```diff
@@ -143,6 +143,9 @@ Metamaps.Visualize = {
       self.mGraph.graph.empty()
     }
 
+
+    if (self.type == 'ForceDirected' && Metamaps.Active.Mapper) $.post('/maps/' + Metamaps.Active.Map.id + '/events/user_presence')
```
Comment (devvmh) [inline app/assets/javascripts/src/Metamaps.Visualize.js]: we should have a better way to say "Map View" or "Topic View"

Maybe something in Metamaps.Settings or Metamaps.Router?

Are there lots of spots in the code that check self.type == ForceDirected?

**Comment by Connoropolous** [comment]
The in progress label just adds itself

File: app/assets/javascripts/src/Metamaps.Visualize.js (hunk 1 above)
Comment (Connoropolous) [inline app/assets/javascripts/src/Metamaps.Visualize.js]: No, it doesn't make that check too many places. The better check might actually be for Metamaps.Active.Map which *should* never be set if there is no active map. And then same with Metamaps.Active.Topic

**Comment by devvmh** [comment]
haha yeah exactly, so is this actually in progress or is it ready?

**Pull request merged**